### DISCUSSION TITLE
Compatibility of fcm_django version 0.3.11 with Django 4.2 and Python 3.11

### DISCUSSION THREAD

**Discussion by LizaMary13** [xtrinch/fcm-django#254] [Q&A]
Hi, I am new to django and using fcm_django.
Can we use legacy API of fcm-django version 0.3.11 with Django 4.2 and python 3.11?
If they're not compatible, what are the steps to follow to upgrade to v2.0.0? I cannot find the fcm module in fcm-django versions after 0.3.11. In a project we had been using the 'fcm_send_bulk_message' function of that module. Has it been removed and what are the alternatives that I can follow?

**Comment by xtrinch** [comment]
No, `fcm-django` `0.3.11` is the old `pyfcm`-based line and it does not explicitly claim compatibility with Django `4.2` / Python `3.11`.

If you need the legacy API, you need to stay on `fcm-django<1`, but compatibility with newer Django/Python versions is not guaranteed there.

If you upgrade to `v1+` or `v2+`, the old `fcm_django.fcm` module is intentionally gone. In the `firebase-admin` based releases:

- `fcm_send_bulk_message(...)` -> `FCMDevice.objects.send_message(Message(...))`
- `fcm_send_message(...)` -> `device.send_message(Message(...))`
- `send_data_message(...)` -> `send_message(Message(data=...))`

Example:

```python
from firebase_admin.messaging import Message, Notification
from fcm_django.models import FCMDevice

FCMDevice.objects.send_message(
    Message(
        notification=Notification(title="title", body="body"),
        data={"key": "value"},
    )
)
```

I also opened PR #301 to add this mapping to the migration docs so future readers can find it directly.
